### ISSUE TITLE
Using less than symbol in version doesn't work  ex. "<3.3.0" updates to "3undefined"

### ISSUE THREAD

**Issue by pborunda** [tjunnone/npm-check-updates#531]
- [x] I have read the list of [known issues](https://github.com/tjunnone/npm-check-updates#known-issues) before filing this issue.
- [x] I have searched for [similiar issues](https://github.com/tjunnone/npm-check-updates/issues) before filing this issue.

* node version: 
v9.3.0

* npm version: 
6.2.0

* npm-check-updates version: 
3.1.7
---------------------------
Using less than symbol in dependency version doesn't work  ex. "<3.3.0" updates to "3undefined".
It should probably stay as "<3.3.0" or resolve to something like "3.2.4". 
I tested with the following
```
$ ncu --packageFile package.json
Checking <my path here>\package.json


 @types/node              11.13.2  →     11.13.4
 jquery                     3.3.1  →       3.4.0
 typescript                <3.3.0  →  3undefined
 webpack                   4.29.6  →      4.30.0
 webpack-bundle-analyzer    3.3.0  →       3.3.2

Run ncu -u to upgrade package.json
```

**Comment by raineorshine** [comment]
Thanks for reporting! I am unable to reproduce, strangely. Do you have a package.json that can reproduce this cleanly? This is what I get:

package.json:

```js
{
  "dependencies": {
    "typescript": "<3.3.0"
  }
}
```

Output:

```sh
$ ncu --packageFile package.json
Checking /private/tmp/temp/package.json
[====================] 1/1 100%

 typescript  <3.3.0  →  ^3.4.3

Run ncu -u to upgrade package.json
```

**Comment by pborunda** [comment]
I tried the above package.json I get the same ^3.4.3. 
But here is a minimum case I see this 3undefined.
```
{
  "devDependencies": {
    "gulp": "^4.0.0",
    "typescript": "<3.3.0",
    "webpack": "4.30.0"
  }
}
```
It looks like if I remove even the "^" from the above or remove the webpack line below, it outputs ok "^3.4.3".

**Comment by raineorshine** [comment]
Great! I can reproduce now. I will investigate.

**Comment by raineorshine** [comment]
Good catch! It was a weird bug in `getPreferredWildcard` which incorrectly returned `undefined` when there  were more version numbers without `^` than with.

Fixed and published in `v3.1.8`.

**Comment by pborunda** [comment]
Thanks!
